packages/shared_preferences/shared_preferences/lib/shared_preferences.dart

```diff
@@ -33,17 +35,23 @@ class SharedPreferences {
   /// previous behavior of this plugin. To use preferences with no prefix,
   /// set [prefix] to ''.
   ///
+  /// If [prefix] is set to '', you may encounter preferences that are
+  /// incompatible with shared_preferences. The optional parameter
+  /// [allowList] will force the plugin to only return preferences that
+  /// are contained in the list AND match the provided prefix.
```

Comment:
We should either be very explicit both here and in the `clear` docs that the allow list doesn't affect `clear`, or we should make it so that it does.

Reply:
Not updating clear feels like a foot gun. But I agree that at minimum the documentation for both should be updated.

Reply:
I do like that there is a clearWithPrefix.

Reply:
the `clear` method shouldn't be used if the `setPrefix` method has been called. I think an assert that is the case is probably worth adding. Thoughts?

Reply:
sorry, I misremembered how that change was handled before. The allowList will be added to clear.